### DISCUSSION TITLE
Updated change log of highlights over the past year?

### DISCUSSION THREAD

**Discussion by AltF5** [winsiderss/systeminformer#1273] [Questions / Answers]
I’ve been out of following the commits & closed issues for a bit, and thus I would be curious to strip out the general fix commits from the shiny new toys. Is someone keeping a general “highlight reel” of features or pressing past issues that go in every month? That changelog.txt might as well be deleted :-)

Features I recently discovered 
- Timeline columns to processes, threads, modules
- other newly added columns in proctree.h (last I had in my copy was the Hex PID) like Image Cohesion, that I suspect is due to jxy’s Process Derpaderp thing 
-  Self-elevate via scheduled task for a quick UAC bypass <== useful!
- Collapsable processes by name <== Wow huge!
- PEView information improvement 
- Column summations

So this begs the question about “what else am I missing? ”… I would be curious of general additions & features at least in the past 24 months if someone has such info. 

Understandable if not and I’ll just go back in time in the coming weeks, but it sure would make life easier :-)

**Comment by Advait251206** [comment]
I don't think there's an official "highlight reel" that summarizes major features over time. Most of the project's history is spread across commits, pull requests, release notes, and discussion threads, which can make it difficult to distinguish major user-facing additions from routine fixes.

From your list, you've already identified several of the more visible improvements that landed over that period:

- Timeline columns for processes, threads, and modules.
- Additional process metadata columns.
- Self-elevation improvements.
- Process tree grouping/collapsing by name.
- PE Viewer enhancements.
- Column aggregation/summation features.

A few places I'd suggest checking if you're trying to catch up on major additions:

1. **GitHub Releases**
   - Release notes tend to contain the most user-visible changes.
   - Even if they're not exhaustive, they're usually a good signal for major features.

2. **Merged Pull Requests**
   - Filtering PRs by label or sorting by popularity can surface larger feature additions that may not stand out in the commit log.

3. **Discussions**
   - Feature requests and announcements often provide more context than the eventual commit message.

4. **Plugins and System Information pages**
   - A surprising number of improvements over the years have landed in analysis tools, ETW/event tracing support, PE inspection, networking, and UI/column enhancements rather than the core process list itself.

I think a periodic "What's New" or "Highlights of the Last Year" post would actually be very valuable. The project moves quickly enough that significant features can easily get lost among hundreds of bug fixes and maintenance commits.

Out of curiosity, are you coming from Process Hacker 2.x or from an older System Informer build? The answer changes quite a bit depending on the starting point, since some of the biggest changes happened during the transition from Process Hacker to System Informer and the subsequent modernization work.

**Reply by stdedos** [reply]
> Out of curiosity, are you coming from Process Hacker 2.x or from an older System Informer build?

I'm coming from https://github.com/winsiderss/systeminformer/releases/tag/v3.2.25011.2103. Idk when I was switched to "Stable" instead of "keeping up with the latest" (I had "some" kernel crash sometime, but I think I moved on).

Also ... from v3.2 I see we are now in v4. Would be nice to know if there is / has been some deprecation (I'm stuck in `[Version 10.0.18363.836]` - and I am super happy 😊)

> I think a periodic "What's New" or "Highlights of the Last Year" post would actually be very valuable

Seconding that! (@dmex)

I understand this is a power tool - so probably I am not using it to it's fullest potential. So for me, a 1y-6mo cadence would be okay. At least "every once a major version bump" 😋

**Reply by Advait251206** [reply]
Thanks for the context—that helps put the gap into perspective. Coming from **v3.2.25011**, you've actually skipped several years of development rather than just a handful of releases, so there are quite a few incremental improvements that never really got summarized in one place.

As far as I'm aware, the jump from v3 to v4 wasn't really about dropping support for older Windows versions or removing lots of functionality. It was more of an evolution of the project, with continued UI refinements, additional inspection capabilities, new columns, PE/ETW improvements, kernel driver work, and general modernization. If you're happily running on Windows 10 1909 (18363), I'd mainly recommend checking the release notes for any version-specific caveats before upgrading.

I still think a "Highlights since v3.2" or "What's New in the Last Year" post would be much more useful than trying to piece everything together from hundreds of commits. There are plenty of features that are easy to miss unless you happen to notice a new column, context menu, or settings option.

Hopefully @dmex considers something like that for future major releases—it would help both long-time users catching up and new users discovering features that otherwise stay hidden.

**Reply by dmex** [reply]
Some highlights - WIP - https://github.com/winsiderss/systeminformer/issues/2943

**Reply by Advait251206** [reply]
Thanks! This is exactly the kind of thing I was hoping for.

Even if it's a WIP, having a curated "highlights" list is much easier to digest than digging through years of commits and closed issues. There are a lot of great features that are easy to miss unless you happen to stumble across them.

If this ends up becoming a recurring document for major releases (or even once every 6–12 months), I think it would be valuable for both long-time users catching up and new users discovering what's available.

Looking forward to #2943!
